wd_helper.au3

```diff
@@ -1048,19 +1048,24 @@ Func _WD_ElementSelectAction($sSession, $sSelectElement, $sCommand, $vParameters
 			"	if (AllowMultiple && SelectElement.multiple == false) {" & _
 			"		return '';" & _
 			"	}" & _
-			"	var options = SelectElement.options;" & _
-			"	var result = false;" & _
-			"	for (var i = 0, o; i < options.length; i++) {" & _
+			"	const LabelsUpperCased = LabelsToSelect.map( function(value) { return value.toUpperCase(); } );" & _ ; https://stackoverflow.com/a/24718430/5314940
+			"	const options = SelectElement.options;" & _
+			"	let result = false;" & _
+			"	for (let i = 0, o, IsDisabled, IsHidden, Matching; i < options.length; i++) {" & _
 			"		o = options[i];" & _
-			"		if (		(LabelsToSelect.indexOf(o.label)!= -1)" & _
-			"				&&	(o.disabled == false	&& (!(o.parentNode.nodeName == 'OPTGROUP' && o.parentNode.disabled)))" & _
-			"				&&	(o.hidden == false	&& (!(o.parentNode.nodeName == 'OPTGROUP' && o.parentNode.hidden)))" & _
-			"			) {" & _
-			"				if (AllowMultiple == false) { SelectElement.selectedIndex = -1; }" & _
-			"				o.selected = true;" & _
-			"				result = true;" & _
-			"				if (AllowMultiple == false) { break; }" & _
-			"			}" & _
+			"		Matching = ( LabelsUpperCased.indexOf( o.label.toUpperCase() ) != -1 );" & _
+			"		if (Matching) {" & _
+			"			IsDisabled =	( o.disabled	|| (o.parentNode.nodeName == 'OPTGROUP' && o.parentNode.disabled) );" & _
+			"			IsHidden =		( o.hidden		|| (o.parentNode.nodeName == 'OPTGROUP' && o.parentNode.hidden) );" & _
+			"			if (IsDisabled || IsHidden) {" & _
+			"					return '';" & _
+			"				} else {" & _
```

Comment:
I thought we agreed that an error wouldn't be generated when disabled or hidden option elements were encountered. This new code would change that.

Reply:
I was thinkinkg about `SINGLESELECT` so will this following modyfication be fine for you ?

```autoit
			"			if (IsDisabled || IsHidden) {" & _
			"					if ( AllowMultiple == false ) { return ''; }" & _
			"				} else {" & _
```

Reply:
Take a look if this [return empty string only on SINGLESELECT](https://github.com/Danp2/au3WebDriver/pull/381/commits/4f8711b4c48ea4b74ec209aa597694ebb5336bd5)

is fine for you.

As in case of `SINGLESELECT` when you try to select DISABLED or HIDDEN <option> element then it will be misleading when you get NOTFOUND instead ELEMENTISSUE

Reply:
That is fine. I've refactored the code further so that it eliminated the unneeded `return` and is easier to follow IMO. Please take a look and provide feedback.

Reply:
I saw your refactor: [Refactor MultiSelectOption JS](https://github.com/Danp2/au3WebDriver/pull/381/commits/8af1743e6f05a2f6d9ade6c281a113b9e4b4dd70)

It looks good, but to be sure did you test it with `DemoSelectOptions` ?
If yes then you can merge.

Reply:
No, I haven't checked with the demo.

Reply:
Ok I will check it ASAP

Reply:
Just checked with this following changes:
[Refactor MultiSelectOption JS](https://github.com/Danp2/au3WebDriver/pull/381/commits/8af1743e6f05a2f6d9ade6c281a113b9e4b4dd70)
[8af1743](https://github.com/Danp2/au3WebDriver/pull/381/commits/8af1743e6f05a2f6d9ade6c281a113b9e4b4dd70)

[Return result from selectAll](https://github.com/Danp2/au3WebDriver/pull/381/commits/e3410191a5ac9d7daa0aa7e87bdad31e73914003)
[e341019](https://github.com/Danp2/au3WebDriver/pull/381/commits/e3410191a5ac9d7daa0aa7e87bdad31e73914003)

[selectAll + $_WD_ERROR_NoMatch](https://github.com/Danp2/au3WebDriver/pull/381/commits/f340a4232380fc12b93770e1f6ac07ef3cbb82b6)

and it is OK.